src/ImageSharp/Processing/Processors/Effects/OilPaintingProcessor{TPixel}.cs

```diff
@@ -52,82 +54,107 @@ protected override void OnFrameApply(ImageFrame<TPixel> source)
 
             int radius = brushSize >> 1;
             int levels = this.definition.Levels;
+            int rowWidth = source.Width;
+            int rectangleWidth = this.SourceRectangle.Width;
 
-            using (Buffer2D<TPixel> targetPixels = this.Configuration.MemoryAllocator.Allocate2D<TPixel>(source.Size()))
-            {
-                source.CopyTo(targetPixels);
+            Configuration configuration = this.Configuration;
 
-                var workingRect = Rectangle.FromLTRB(startX, startY, endX, endY);
-                ParallelHelper.IterateRows(
-                    workingRect,
-                    this.Configuration,
-                    rows =>
-                        {
-                            for (int y = rows.Min; y < rows.Max; y++)
-                            {
-                                Span<TPixel> sourceRow = source.GetPixelRowSpan(y);
-                                Span<TPixel> targetRow = targetPixels.GetRowSpan(y);
+            using Buffer2D<TPixel> targetPixels = this.Configuration.MemoryAllocator.Allocate2D<TPixel>(source.Size());
+            source.CopyTo(targetPixels);
 
-                                for (int x = startX; x < endX; x++)
-                                {
-                                    int maxIntensity = 0;
-                                    int maxIndex = 0;
+            var workingRect = Rectangle.FromLTRB(startX, startY, endX, endY);
+            ParallelHelper.IterateRows(
+                workingRect,
+                this.Configuration,
+                (rows) =>
+                {
+                    // Allocate the reusable source row buffer, to enable vectorized bulk conversions
+                    using IMemoryOwner<Vector4> sourceRowBuffer = configuration.MemoryAllocator.Allocate<Vector4>(rowWidth);
```

Comment:
I'm not a big fan of the new `using` declaration as I believe it reduces readability in many situations and can promote maintaining object lifespans for longer than they should be. 

If they were to be used (haha) then I would restrict the environment to small methods with a single instance. That is not possible however so I'm very tempted to add a rule that disallows them.

Reply:
So let the fight begin! Me and @brianpopow like them 😄

Reply:
I have to say I like them very much too, though I can see how in some cases they might make the code less intuitive at first. For now I've removed them in https://github.com/SixLabors/ImageSharp/pull/1068/commits/c89e0b1e5a1ff9e266cb3a8ef5febaabbde13059, and if we change our mind on this we can always refactor this later on, it's just a matter of removing two brackets anyway 👍